### PR TITLE
update branch 0.10.x

### PR DESCRIPTION
**Pull request by alacuku**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

 /kind bug

/kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

> /area driver-bpf

> /area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap-engine-udig

> /area libscap

/area libpman

/area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
This PR cherrypicks the commits introduced by the following PRs:
* #888 
* #871
* #894 
* #895

to `release/0.10.x` branch.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: userspace/libscap/scap_savefile.c [hunk 1]
```diff
@@ -541,15 +542,12 @@ static int32_t scap_write_proclist(scap_t *handle, scap_dumper_t *d)
...
-	//
-	// Exit immediately if the process list is empty
-	//
-	if(HASH_COUNT(handle->m_proclist.m_proclist) == 0)
+	scap_dumper_t *proclist_dumper = scap_write_proclist_begin(handle);
+	if(proclist_dumper == NULL)
 	{
-		return SCAP_SUCCESS;
+		return SCAP_FAILURE;
 	}
```
Comment (LucaGuerra) [inline userspace/libscap/scap_savefile.c:549]: this is a bit different: https://github.com/falcosecurity/libs/pull/888/files#diff-7b4e2af897ce4b8645f4f862762f68425d8394805c45c194255730e99078767dR717

probably needs to be fixed

Comment (alacuku) [reply userspace/libscap/scap_savefile.c:549]: That change is not present in the `release/0.10.x` branch: https://github.com/falcosecurity/libs/blame/release/0.10.x/userspace/libscap/scap_savefile.c#L552
We need to keep it this way. Or we backport the commit that introduced that change into `release/0.10.x` branch.

Comment (FedeDP) [reply userspace/libscap/scap_savefile.c:549]: Do we miss the
```
	//
	// Exit immediately if the process list is empty
	//
	if(HASH_COUNT(handle->m_proclist.m_proclist) == 0)
	{
		return SCAP_SUCCESS;
	}
```
check now? It was present back then in 0.10.x.

**Review by FedeDP — COMMENTED** [review]
Left a comment, otherwise this LGTM.

File: userspace/libscap/scap_savefile.c (hunk 1 above)
Comment (alacuku) [reply userspace/libscap/scap_savefile.c]: :+1:

**Review by FedeDP — APPROVED** [review]
/approve

**Review by LucaGuerra — APPROVED** [review]

**Pull request merged**